### PR TITLE
Underline the word right under it

### LINKED ISSUE #164: Proposal: highlight under command
# Before
screen shot 2016-11-28 at 23 19 58
# After
screen shot 2016-11-28 at 23 19 46

I modified the code and wondered if this is cool or not?

### PR DESCRIPTION
**Pull request by stoivo**
closes #164

I want to get it out there. what do you think?

### CODE REVIEW

**Comment by stoivo** [comment]
And why do we only draw lines to the 20 first ones?

**Comment by idank** [comment]
I'm not sure about the 20 lines. Want to open a new issue and add a reproduction there?

I'll look at the pull request soon, thanks.

**Comment by stoivo** [comment]
I added a new PR for that Mark all flags #167,

Totally independent.

**Comment by idank** [comment]
I tried reviewing the code but it's hard because you renamed variables in addition to the change you did. I'd prefer it if you didn't selectively start renaming variables because it's weird when the ones you touched are the only ones with a long name with underscores. Can you add comments instead? Thanks.

**Comment by stoivo** [comment]
I will make some smaller easy to read commit when I have the time.

**Comment by stoivo** [comment]
Please try to review it now, commit by commit.

File: explainshell/web/static/js/es.js [hunk 1]
```diff
@@ -563,20 +563,21 @@ function drawgrouplines(commandselector, options) {
...
-        var rr = link.option.getBoundingClientRect(),
-            spanmid = rr.left + rr.width / 2,
-            rrright = rr.right - strokewidth;
 
-        var path = new espath();
 
-        link.paths.push(new espath().addpoint(rr.left, 0).addpoint(rrright, 5));
-        link.paths.push(new espath().addpoint(rrright, 5).addpoint(rrright, 0));
-        path.addpoint(rr.left + rr.width / 2, 5); // 3
+        var command = link.option.getBoundingClientRect(),
```
Comment (idank) [inline explainshell/web/static/js/es.js]: commandRect?

**Comment by idank** [comment]
Thanks, much easier. Since you're renaming some variables, let's start using camelCase.

File: explainshell/web/static/js/es.js [hunk 2]
```diff
@@ -669,10 +673,17 @@ function drawgrouplines(commandselector, options) {
...
-            link.paths.push(new espath().addpoint(rr.left, 0).addpoint(rrright, 5));
-            link.paths.push(new espath().addpoint(rrright, 5).addpoint(rrright, 0));
+            link.paths.push(new espath()
+              .addpoint(rr.left, 0)
+              .addpoint(rr.left, 5)
+              .addpoint(rrright, 5)
+              .addpoint(rrright, 0)
+            );
             var rrmid = d3.round(rr.left + rr.width / 2);
-            link.lines.push({x1: rrmid, y1: 6, x2: rrmid, y2: 5+unknownlinelength+command_offsett_to_canvas});
```
Comment (idank) [inline explainshell/web/static/js/es.js]: Any reason you changed this?

File: explainshell/web/static/js/es.js [hunk 3]
```diff
@@ -572,6 +572,7 @@ function drawgrouplines(commandselector, options) {
             command_right = command.right - strokewidth;
 
         link.starty = command.bottom - command_wrapper.top + 1;
+        command_offsett_to_canvas = margin_between_command_and_canvas - link.starty
```
Comment (idank) [inline explainshell/web/static/js/es.js]: offset

File: explainshell/web/static/js/es.js (hunk 2 above)
Comment (stoivo) [inline explainshell/web/static/js/es.js]: Nothing else than using the the same pattern. This is only for me to easily understand.

Revert it?

File: explainshell/web/static/js/es.js (hunk 1 above)
Comment (stoivo) [inline explainshell/web/static/js/es.js]: I will do

File: explainshell/web/static/js/es.js (hunk 3 above)
Comment (stoivo) [inline explainshell/web/static/js/es.js]: oh, thanks

**Comment by stoivo** [comment]
Hello, would you take the time to look at this?

**Comment by idank** [comment]
I'm away from home for a while but I'll get to it when I can. Sorry for the
delay.

**Comment by stoivo** [comment]
Yes, how are we doing?

**Comment by idank** [comment]
I haven't forgotten. Hopefully will get to it this weekend. Thanks for your patience!

File: explainshell/web/static/js/es.js [hunk 4]
```diff
@@ -573,10 +573,16 @@ function drawgrouplines(commandselector, options) {
             commandRight = commandRect.right - strokewidth;
 
         link.starty = commandRect.bottom - commandWrapperRect.top + 1;
+        commandOffsettToCanvas = marginBetweenCommandAndCanvas - link.starty
```
Comment (idank) [inline explainshell/web/static/js/es.js]: commandOffsettToCanvas -> commandOffsetToCanvas

Comment (idank) [inline explainshell/web/static/js/es.js]: also, is this missing a 'var'?

File: explainshell/web/static/js/es.js [hunk 5]
```diff
@@ -568,15 +568,23 @@ function drawgrouplines(commandselector, options) {
...
+          .addpoint(commandRect.left, 0)
+          .addpoint(commandRect.left, 5)
+          .addpoint(commandRight, 5)
+          .addpoint(commandRight, 0)
+        );
 
-        link.paths.push(new espath().addpoint(rr.left, 0).addpoint(rrright, 5));
-        link.paths.push(new espath().addpoint(rrright, 5).addpoint(rrright, 0));
-        path.addpoint(rr.left + rr.width / 2, 5); // 3
+        var path = new espath();
```
Comment (idank) [inline explainshell/web/static/js/es.js]: what's going on here? is 'path' used anywhere after this?

Comment (stoivo) [inline explainshell/web/static/js/es.js]: Yes we continue to add points here
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR602
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR605
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR617
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR627
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR630
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR642

and then we add path to link
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR646

**Comment by stoivo** [comment]
Nice, I fixed the typo and commented your comment.

**Comment by idank** [comment]
Hey, I was trying this today to see how it looks and saw [this weird glitch](http://i.imgur.com/PXzcEKc.png), any idea why it's messed up like that?

**Comment by idank** [comment]
The command I used was

    tar zcf - some-dir | ssh some-server "cd /; tar xvzf -" | tar zcf - some-dir | ssh some-server "cd /; tar xvzf -" | tar zcf - some-dir | ssh some-server "cd /; tar xvzf -"

**Comment by stoivo** [comment]
Interesting, I'll take a look

**Comment by stoivo** [comment]
I think this explains it 
screen shot 2017-04-07 at 19 14 08

**Comment by stoivo** [comment]
I need to tune 
https://github.com/stoivo/explainshell/blob/151897a9acb6d3eae0da28e13f0b20ad36d630b7/explainshell/web/static/js/es.js#L612-L617
https://github.com/stoivo/explainshell/blob/151897a9acb6d3eae0da28e13f0b20ad36d630b7/explainshell/web/static/js/es.js#L637-L642

**Comment by stoivo** [comment]
I fixed it :tada:  I added couple of screenshots. 
screen shot 2017-04-07 at 21 19 12
screen shot 2017-04-07 at 21 19 54
screen shot 2017-04-07 at 21 19 29

I didn't know what this feature existed. Thank for feedback. You can test it again.

**Comment by stoivo** [comment]
Ping

**Comment by idank** [comment]
Will do my best this weekend. Thanks for your patience, it's admirable. :)

**Comment by idank** [comment]
I'm trying to review this but honestly I can't remember what the code does and it's hard to know if anything about the existing drawing algorithm gets broken.

I think I'll pull this locally and run with it for a few days for my own personal use. If I don't see anything weird I'll take it.

Eventually I'd really like to rewrite the drawing algorithm so it supports multi line commands, is better documented and is easier to modify.

**Comment by stoivo** [comment]
Soinds good, it was kind of weird logic

**Comment by stoivo** [comment]
How are we doing?

**Comment by idank** [comment]
Didn't see anything unusual. I'm away on a trip, will push it in a few days.

**Comment by stoivo** [comment]
Seen any bugs?

**Comment by idank** [comment]
None that I could see, so I finally pushed it! Congrats and thank you!

I'm sorry it took this long. I've been really busy with other things and I wish I had more time to work on the site, I have so many ideas in the back of my head, somewhere. But it makes me happy that people like you care about the site and continue improving it.

If you'd like, I'll be happy to send you a little token of appreciation for your contributions to the site (this isn't your first!). Reach me out privately at idan@explainshell.com and we can sort out the details.

**Pull request closed without merging**